modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java

```diff
@@ -422,6 +423,22 @@ public ObjectDefinition putObjectDefinition(
 				_objectDefinitionService.getObjectDefinition(
 					objectDefinitionId);
 
+		String rootObjectDefinitionExternalReferenceCode =
+			objectDefinition.getRootObjectDefinitionExternalReferenceCode();
+
+		_validateObjectDefinitionStatus(
+			objectDefinition.getStatus(),
+			serviceBuilderObjectDefinition.getStatus());
+
+		if (Validator.isNotNull(rootObjectDefinitionExternalReferenceCode) &&
+			FeatureFlagManagerUtil.isEnabled("LPS-187142")) {
```

Comment:
@G-Sa reorder since `rootObjectDefinitionExternalReferenceCode` is not needed for `_validateObjectDefinitionStatus`

```
_validateObjectDefinitionStatus(
	objectDefinition.getStatus(),
	serviceBuilderObjectDefinition.getStatus());

String rootObjectDefinitionExternalReferenceCode =
	objectDefinition.getRootObjectDefinitionExternalReferenceCode();

if (Validator.isNotNull(rootObjectDefinitionExternalReferenceCode) &&
	FeatureFlagManagerUtil.isEnabled("LPS-187142")) {
```